### PR TITLE
feat: Billing limit and billing info api

### PR DESCRIPTION
**Pull request by tiina303**
1. Sending up to billing limit events count to stripe.
2. Backend changes for https://github.com/PostHog/posthog/pull/9304
Screen Shot 2022-04-04 at 20 36 46

### CODE REVIEW

File: multi_tenancy/views.py [hunk 1]
```diff
@@ -73,6 +77,26 @@ def get_object(self) -> OrganizationBilling:
...
+        current_bill = get_current_usage_bill(
+            serializer.instance.stripe_subscription_id
+        )
+        proposed_new_limit = self.request.data["billing_limit"]
+        if current_bill > proposed_new_limit:
+            if settings.DEBUG:
+                print(
+                    f"Can't set billing limit to be {proposed_new_limit} as it's lower than current bill {current_bill}"
+                )
+            return
```
Comment (timgl) [inline multi_tenancy/views.py:91]: Might be worth returning an error here which should automatically be displayed in the frontend

File: multi_tenancy/stripe.py [hunk 2]
```diff
@@ -165,6 +176,16 @@ def compute_webhook_signature(payload: str, secret: str) -> str:
     return stripe.webhook.WebhookSignature._compute_signature(payload, secret)
 
 
+def set_billing_limit_for_customer(customer_id: str, billing_limit_usd: int) -> bool:
+    _init_stripe()
+    try:
+        stripe.Customer.modify(
+            customer_id, metadata={"billing_limit": billing_limit_usd}
+        )
+    except Exception as e:
```
Comment (timgl) [inline multi_tenancy/stripe.py:185]: Why capture the exception here? It'll mean that we don't retry

File: multi_tenancy/models.py [hunk 3]
```diff
@@ -203,6 +219,36 @@ def handle_post_card_validation(self) -> "OrganizationBilling":
...
+        billing_limit_max_events = get_billing_limit_events_count(
+            self.stripe_subscription_id, self.billing_limit
+        )
+        print(f"{current_events_count} {billing_limit_max_events}")
+        if current_events_count is None or billing_limit_max_events is None:
+            return event_count  # Some information is badly formatted - ignore limiting
+        if current_events_count + event_count > billing_limit_max_events:
+            event_count = max(billing_limit_max_events - current_events_count, 0)
+            if not self.billing_limit_exceeded:
+                self.billing_limit_exceeded = True
```
Comment (timgl) [inline multi_tenancy/models.py:242]: Definitely worth sending an email when this happens. See posthog/api/authentication.py for an example.

Comment (tiina303) [reply multi_tenancy/models.py:242]: To the user you mean, right?

File: multi_tenancy/views.py (hunk 1 above)
Comment (tiina303) [reply multi_tenancy/views.py:91]: oh sweet didn't know that's possible

File: multi_tenancy/models.py (hunk 3 above)
Comment (timgl) [reply multi_tenancy/models.py:242]: Yep!

Comment (tiina303) [reply multi_tenancy/models.py:242]: will do that in a follow-up PR this one's already big

**Comment by tiina303** [comment]
The main repo tests are failing, but they also fail on a [no-op pr](https://github.com/PostHog/posthog-cloud/runs/6116800902?check_suite_focus=true), so they are just broken

**Pull request merged**